server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -393,9 +417,41 @@ public DateFieldMapper build(MapperBuilderContext context) {
                 nullTimestamp,
                 resolution,
                 context.isSourceSynthetic(),
+                indexMode,
+                indexSortConfig,
+                hasDocValuesSparseIndex,
                 this
             );
         }
+
+        private boolean hasInvertedIndex(boolean hasDocValuesSparseIndex) {
+            return index.isConfigured()
+                ? index.getValue() && indexCreatedVersion.isLegacyIndexVersion() == false
+                : hasDocValuesSparseIndex == false;
+        }
+
+        private static boolean hasDocValuesSparseIndex(
+            final Parameter<Boolean> index,
+            final IndexVersion indexCreatedVersion,
+            final String fullFieldName,
+            final IndexMode indexMode,
+            final IndexSortConfig indexSortConfig,
+            final boolean hasDocValues
+        ) {
+            if (index.isConfigured()) {
```

Comment:
This does not work in case `index: true` explicitly because `isSet` is `true` but `getValue() == getDefaultValue()`.

Reply:
In case of `@timestamp` for `isConfigured` we have two possibilities:
1. `isConfigured()` is `true` which means the user explicitly set the `index` value to `true` or `false`. In such case no matter the value of `index` the sparse doc values index should be disabled and we would use the inverted index
2. `isConfigured()` is `false`: this means the `index` parameter is not set by default which means for LogsDB under certain conditions we can use the sparse index instead of the inverted index.

Reply:
Maybe we should then just check for isSet() ?

Reply:
If we use `iseSet` than the other test fails...for instance `testFieldTypeWithSkipDocValues_LogsDBMode`